python_src/src/lamp_py/performance_manager/gtfs_utils.py

```diff
@@ -49,3 +63,184 @@ def start_time_to_seconds(
         return time
     (hour, minute, second) = time.split(":")
     return int(hour) * 3600 + int(minute) * 60 + int(second)
+
+
+def add_fk_static_timestamp_column(
+    events_dataframe: pandas.DataFrame,
+    timestamp_column: str,
+    db_manager: DatabaseManager,
+) -> pandas.DataFrame:
+    """
+    adds "fk_static_timestamp" column to dataframe
+
+    using "fk_static_timestamp" column, events dataframe records may be joined to
+    gtfs static record tables
+    """
+    process_logger = ProcessLogger(
+        "add_fk_static_timestamp",
+        row_count=events_dataframe.shape[0],
+        timestamp_column=timestamp_column,
+    )
+    process_logger.log_start()
+
+    # get unique "start_date" values from events dataframe
+    # with associated minimum value of timestamp column
+    date_groups = events_dataframe.groupby(by="start_date")[
+        timestamp_column
```

Comment:
Basically a copy and paste from the functions that used to be in `l0_rt_trip_updates.py` and `l0_rt_vehicle_positions.py`. 

Now this only adds the `fk_static_timestamp` column, the adding of the `parent_station` column occurs in separate function.